discovery_engine_core/core/src/engine.rs

```diff
@@ -457,6 +464,66 @@ where
         }
     }
 
+    /// Perform an active search by topic.
+    pub async fn search_by_topic(
+        &mut self,
+        topic: &str,
+        page: u32,
+        page_size: u32,
+    ) -> Result<Vec<Document>, Error> {
+        if topic.trim().is_empty() {
```

Comment:
🌱 Except for the query, both methods ( `active_search` and `search_by_topic`) share the same code. Maybe it makes sense to refactor the code a bit in a follow-up future.